### PR TITLE
Enabling comparison of multiple jacobians

### LINKED ISSUE #486: Display info from multiple Jacobian options
Allow multiple Jacobian options to be set in the ini file.  These should then be communicated through the table and performance profiler, just as a minimizer would be.

Possibly, we should have, for example:
lm-scipy:analytic
lm-scipy:2-point
lm-scipy:3-point
etc, as columns in the benchmarking table.

### PR DESCRIPTION
**Pull request by wathen**
#### Description of Work

Fixes #486 

Adds the functionality to test minimizers with multiple Jacobians and view as extra columns in the results tables.

#### Testing Instructions

1. Tables are correctly generated with Jacobian names stated in the columns 

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: docs/source/extending/controllers.rst [hunk 1]
```diff
@@ -42,8 +42,15 @@ In order to add a new controller, you will need to:
                          'scipy_ls' does but the derivative is handle internally
                          within the sofware package)
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/extending/controllers.rst:43]: Done

File: docs/source/extending/controllers.rst [hunk 2]
```diff
@@ -42,8 +42,15 @@ In order to add a new controller, you will need to:
...
         - 'general' - minimizers which solve a generic `min f(x)`
-      and the values as a list of minimizer for that specific controller that fit into
-      each of the above categories. See for example the ``GSL`` controller.
+      and the values as a list of minimizer for that specific controller that fit into each of the above categories. See for example the ``GSL`` controller.
+   -  ``jacobian_information()``: Setups up Jacobian information for the
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/extending/controllers.rst:46]: corrected

File: docs/source/extending/controllers.rst [hunk 3]
```diff
@@ -42,8 +42,15 @@ In order to add a new controller, you will need to:
...
+      controller. This should return the following arguments:
+
+      - ``has_jacobian``: a True or False value whether the controller
+        requires Jacobian information
+      - ``jacobian_list``: a list of minimizers in a specific software
+        that do not require Jacobian informations. For example in the
+        ``ScipyLS`` controller this would return ``lm-scipy-no-jac``
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/extending/controllers.rst:53]: corrected

File: fitbenchmarking/controllers/base_controller.py [hunk 4]
```diff
@@ -155,6 +157,20 @@ def check_attributes(self):
...
+    def jacobian_information(self):
+        """
+        Setups up Jacobian information for the controller. This should return the following arguments:
+
+        has_jacobian: a True or False value whether the controller
+        requires Jacobian information
+
+        jacobian_list: a list of minimizers in a specific software
+        that do not require Jacobian informations. For example in the
+        `ScipyLS` controller this would return `lm-scipy-no-jac`
```
[the remark this answers is not in the document]
Comment (wathen) [reply fitbenchmarking/controllers/base_controller.py:170]: corrected

File: docs/source/users/options/fitting_option.rst [hunk 5]
```diff
@@ -101,23 +101,3 @@ Default is ``SciPyFD``
 
     [FITTING]
     jac_method: SciPyFD
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/users/options/fitting_option.rst:103]: corrected

File: docs/source/extending/jacobian_extend.rst [hunk 6]
```diff
@@ -8,14 +8,15 @@ Adding new Jacobians
 
 In order to add a new Jacobian evaluation method, you will need to:
 
-1. Give the Jacobian a name using the following convention ``<jac_method>`` and
-   ``<num_method>``. An example could be ``scipy_fd`` for ``<jac_method>`` and
-   ``2point`` for ``<num_method>`` which would call the SciPy's 2-point finite
-   difference approximation.
+1. Give the Jacobian method in the :ref:`fitting_option` using the following
+   convention ``<jac_method>`` and :ref:`jacobian_option` is added using
+   ``<jac_method>`` as the key and the list as possible numerical methods.
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/extending/jacobian_extend.rst:13]: Agreeded

File: docs/source/extending/jacobian_extend.rst [hunk 7]
```diff
@@ -8,14 +8,15 @@ Adding new Jacobians
...
 
-2. Create ``fitbenchmarking/jacobian/<jac_method>_<num_method>_jacobian.py``
+2. Create ``fitbenchmarking/jacobian/<jac_method>_jacobian.py``
    which contains a new subclass of ``Jacobian``
-   (from ``base_jacobian.py``).
+   (from ``base_jacobian.py``). The numerical method is set sequentially
+   withing ``fitbenchmarking.core.fitting_benchmark.loop_over_jacobians`` by
+   using the ``method`` attribute of the class.
    Then implement the method ``eval()``, which evaluates the Jacobian.
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/extending/jacobian_extend.rst:20]: I'm not 100% sure what you mean here?

File: docs/source/extending/controllers.rst [hunk 8]
```diff
@@ -36,19 +36,23 @@ In order to add a new controller, you will need to:
...
         - 'deriv_free' - derivative free algorithms (these are algorithms that
-                         do not require an information about derivatives. For
-                         example, the 'Simplex' method in 'Mantid' does not
-                         require derivative information but 'lm-scipy-no-jac' in
-                         'scipy_ls' does but the derivative is handle internally
-                         within the sofware package)
+          do not require an information about derivatives. For example, the
+          'Simplex' method in 'Mantid' does not require derivative information
+          but 'lm-scipy-no-jac'in 'scipy_ls' does but the derivative is handle
+          internally within the sofware package)
```
Comment (tyronerees) [inline docs/source/extending/controllers.rst:42]:
```suggestion
          cannot use derivative information. For example, the
          'Simplex' method in 'Mantid' does not require Jacobians.
          However, 'lm-scipy-no-jac' in 'scipy_ls' is designed to use derivatives, 
          but calculates an approximation internally if one is not supplied.)
```

File: docs/source/extending/controllers.rst [hunk 9]
```diff
@@ -36,19 +36,23 @@ In order to add a new controller, you will need to:
...
       controller. This should return the following arguments:
 
       - ``has_jacobian``: a True or False value whether the controller
+<<<<<<< HEAD
         requires Jacobian information
+      - ``jacobian_free_solvers``: a list of minimizers in a specific software
+=======
+        can use Jacobian information
       - ``jacobian_list``: a list of minimizers in a specific software
+>>>>>>> 1221ec24efec68e1faf69c2e282f25155916176b
```
Comment (tyronerees) [inline docs/source/extending/controllers.rst]:
```suggestion
        requires Jacobian information
      - ``jacobian_free_solvers``: a list of minimizers in a specific software
```

**Comment by wathen** [comment]
I think I have done all the things you spotted

**Pull request merged**